vallie/vallie.go

```diff
@@ -0,0 +1,30 @@
+// Package vallie stands in for a stand-alone validation library
+// purely for discussion purposes.
+//
+// Assuming this is a desired approach, more fidelity will be added
+// and vallie would be extracted into it's own repo
+package vallie
+
+import (
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+var errZeroVal = errors.New("Object is a zero value")
+
+// Validate accepts objects to be validated from constructor and returns an error
+// if any of the provided object is a zero value object
+//
+// TODO: if Validate returns an error, how can we hook into the dig resolution
```

Comment:
What about actionable results? E.g. return 2 things: a constructor for unrecognized types and an error, so a user can simply register it if they want. Another option would be to return a slice of types for a user to set(may be even other than a zero value) and register manually.